rust-toolchain

```diff
@@ -1 +1 @@
-1.64.0
+1.66.0
```

Comment:
This is in sync with upstream https://github.com/privacy-scaling-explorations/halo2/blob/main/rust-toolchain#L1 :+1: